zmon_worker_monitor/builtins/plugins/kubernetes.py

```diff
@@ -0,0 +1,420 @@
+            if (replicas is None or deployment.replicas == replicas) and (ready is None or deployment.ready == ready)
+        ]
+
+    def cm(self, **kwargs):
+        """Alias for configmaps"""
+        return self.configmaps(**kwargs)
+
+    def configmaps(self, name=None, **kwargs):
+        """
+        Return list of ConfigMaps.
+
+        :param name: ConfigMap name.
+        :type name: str
+
+        :param **kwargs: ConfigMap labelSelector filters.
+        :type **kwargs: dict
+
+        :return: List of ConfigMaps. Typical ConfigMap has "metadata", "status" and "spec".
+        :rtype: list
+        """
+        filter_kwargs = self._get_filter_kwargs(name=name, **kwargs)
+
+        query = pykube.ConfigMap.objects(self.__client).filter(**filter_kwargs)
+
+        configmaps = self._get_resources(query)
+
+        return [configmap.obj for configmap in configmaps]
+
+    def pvc(self, **kwargs):
+        """Alias for persistentvolumeclaims"""
+        return self.persistentvolumeclaims(**kwargs)
+
+    def persistentvolumeclaims(self, name=None, phase=None, **kwargs):
+        """
+        Return list of PersistentVolumeClaims.
+
+        :param name: PersistentVolumeClaim name.
+        :type name: str
+
+        :param phase: Volume phase.
+        :type phase: str
+
+        :param **kwargs: PersistentVolumeClaim labelSelector filters.
+        :type **kwargs: dict
+
+        :return: List of PersistentVolumeClaims. Typical PersistentVolumeClaim has "metadata", "status" and "spec".
+        :rtype: list
+        """
+        filter_kwargs = self._get_filter_kwargs(name=name, **kwargs)
+
+        query = pykube.PersistentVolumeClaim.objects(self.__client).filter(**filter_kwargs)
+
+        pvcs = self._get_resources(query)
+
+        return [pvc.obj for pvc in pvcs if phase is None or pvc.obj['status'].get('phase') == phase]
+
+    def pv(self, **kwargs):
+        """Alias for persistentvolumes"""
+        return self.persistentvolumes(**kwargs)
+
+    def persistentvolumes(self, name=None, phase=None, **kwargs):
+        """
+        Return list of PersistentVolumes.
+
+        :param name: PersistentVolume name.
+        :type name: str
+
+        :param phase: Volume phase.
+        :type phase: str
+
+        :param **kwargs: PersistentVolume labelSelector filters.
+        :type **kwargs: dict
+
+        :return: List of PersistentVolumes. Typical PersistentVolume has "metadata", "status" and "spec".
+        :rtype: list
+        """
+        filter_kwargs = self._get_filter_kwargs(name=name, **kwargs)
+
+        query = pykube.PersistentVolume.objects(self.__client).filter(**filter_kwargs)
+
+        # PersistentVolume does not belong to a "namespace". Calling with **default** will fail with 404.
+        pvs = query.all()
+
+        return [vc.obj for vc in pvs if phase is None or vc.obj['status'].get('phase') == phase]
+
+    def metrics(self):
+        """
+        Return cluster wide metrics in prometheus format.
```

Comment:
I think I may have caused some confusion, this is not cluster wide metrics, but only metrics for the apiserver.